fdbserver/DataDistribution.actor.cpp

```diff
@@ -4486,6 +4514,45 @@ ACTOR Future<Void> ddSnapCreate(DistributorSnapRequest snapReq, Reference<AsyncV
 	return Void();
 }
 
+ACTOR Future<Void> ddExclusionSafetyCheck(DistributorExclusionSafetyCheckRequest req,
+                                          Reference<DataDistributorData> self, Database cx) {
+	TraceEvent("DDExclusionSafetyCheckBegin");
+	vector<StorageServerInterface> ssis = wait(getStorageServers(cx));
```

Comment:
Move this below `if (!self->teamCollection)` to avoid unnecessary work in case `teamCollection` is null.

Reply:
It can't be in between the null check and usage of the teamCollection, because the wait creates a context switch which can invalidate the null check's result.